pkg/lib/cockpit-components-table.tsx

```diff
@@ -266,24 +271,26 @@ export const ListingTable = ({
 
         const rowKey = rowProps.key || rowIndex;
         const isExpanded = expanded[rowKey] === undefined ? !!row.initiallyExpanded : expanded[rowKey];
+        if (isExpandable) {
+            rowProps.isContentExpanded = Boolean(row.expandedContent) && isExpanded;
+        }
         let columnSpanCnt = 0;
         const rowPair = (
             <React.Fragment key={rowKey + "-inner-row"}>
                 <Tr {...rowProps}>
                     {isExpandable
                         ? (row.expandedContent
-                            ? <Td expand={{
-                                // HACK - rowIndex is declared to be a number, but we have always used
-                                //        it with strings, and that worked ok. The tests expect it.
-                                //        But I guess we really shouldn't...
-                                rowIndex: rowKey as number,
-                                isExpanded,
-                                onToggle: () => {
-                                    if (afterToggle)
-                                        afterToggle(!expanded[rowKey]);
-                                    setExpanded({ ...expanded, [rowKey]: !expanded[rowKey] });
-                                }
-                            }} />
+                            ? <Td
+                                data-ouia-component-id={rowKey.toString()}
+                                expand={{
+                                    rowIndex,
+                                    isExpanded,
+                                    onToggle: () => {
+                                        if (afterToggle)
+                                            afterToggle(!expanded[rowKey]);
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22332-cfbd59ab-20250819-153747-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)